### ISSUE TITLE
Recommended papers is empty

### ISSUE THREAD

**Issue by Randl** [karpathy/arxiv-sanity-preserver#57]
I've registered a week or so ago, added some papers to library, but the recommended papers tab is empty

**Comment by seandoyle** [comment]
I've been using it for months (it's wonderful) but the last 3 weeks or so I'm not getting new papers. I can still get earlier papers though - have you tried setting the retrieval window to be 1 month or more?

**Comment by karpathy** [comment]
@seandoyle can you clarify what the issue is? I'm not able to reproduce any problems with my own account, and I have a number of recommended papers over last week/month etc.

**Comment by seandoyle** [comment]
I'm not sure what the issue is - here's what I observe:
- When I select recommended for one week I get zero hits.  I've been getting zero hits for the last few weeks when I use the 1 week window.
- But when I click on last month - I get a bunch - including one from 1/24. Since today is 1/31 - I would have expected this to show up sometime in the last week but I don't remember seeing it.
- The frequency of papers really fell off starting about 3 week ago. Most of what is in my library is medical image CNNs or segmentation which are pretty active areas.
- When I use the one month window - there are 8 papers. 5 of them are on the dates 1/3, 1/4, and 1/5. This is an odd clustering of dates. 

So - maybe this is just some type of random occurrence. Or maybe the selection of papers in my library is excluding papers in some way. 

If there is something I should try here (deleting/adding papers &etc) I'd be happy to try. This is such a wonderful resource & I really appreciate you making this available.

**Comment by karpathy** [comment]
I see, I think this is related to the fact that I forget to log in and recompute arxiv-sanity. I have to do this manually every ~2-3 days right now and sometimes I forget. I'm going to move arxiv-sanity to a more permanent location soon and these issues will go away.

**Comment by karpathy** [comment]
btw I finished the migration today and this should never happen again. If it does, definitely a bug.

**Comment by seandoyle** [comment]
Works perfectly now. Thank you - this is wonderful.

**Comment by Randl** [comment]
Looks like I'm facing the issue once again

**Comment by seandoyle** [comment]
Yes - I'm having the same symptom again. I don't have any new papers in the last few weeks.  In early April I added a few more papers to my library and the total was ~40. I deleted some of these yesterday in case I had hit a limit or there was something about those papers that was causing the algorithm to fail - but this doesn't seem to be the issue.

**Comment by Randl** [comment]
ping @karpathy

**Comment by karpathy** [comment]
sorry, i was on a vacation and memory started exploding on the machine for some reason that I think has to do with mongodb taking over the RAM. I'll have to figure it out later. Resolving now with a hack by making the codes a bit smaller... Should be up and functional again in ~2hrs

**Comment by Randl** [comment]
@karpathy sorry to bother, but looks like it's happening once again. For "top resent" too

**Comment by karelin** [comment]
@Randl Have you tried to switch to different periods (say, last month or last year). In may case, last day and last 3 days are empty, too, but I don't think it is a bug

**Comment by Randl** [comment]
@karelin I've never had empty 3-day period except when the bug occurred. The fact that yours is empty too is additional evidence. 

Obviously for one-month period there will be papers until one month passes :)

**Comment by karpathy** [comment]
I think this is happening because no arxiv papers get published over the weekend. So "last 3 days" ends up empty.

**Comment by karelin** [comment]
I have an idea for this: as the site (both recommendations and main data base) are updated 'offline', perhaps the header of the main page could contain the "last updated" dates?

**Comment by Randl** [comment]
@karpathy  Well, today is Tuesday, so I suppose that's not the reason

**Comment by Randl** [comment]
@karpathy Just want to note that the same thing still happens from time to time (i.e. today), and last week too

**Comment by Randl** [comment]
After launching a local instance, I've noted that sometimes when I try to update PDFs, arxiv bans me (makes sense given I downloaded order of 100000 papers from there). If this indeed the case, maybe just stopping update if a number of downloads failed and rerunning it in an hour can solve the issue
